pkg/sip/inbound.go

```diff
@@ -1062,17 +1080,49 @@ func (c *inboundCall) closeWithNoACK() {
 }
 
 func (c *inboundCall) closeWithCancelled() {
-	c.state.DeferUpdate(func(info *livekit.SIPCallInfo) {
-		info.DisconnectReason = livekit.DisconnectReason_CLIENT_INITIATED
-	})
-	c.close(false, CallHangup, "cancelled")
+	var reason ReasonHeader
+	if p := c.closeReason.Load(); p != nil {
+		reason = *p
+	}
+	c.closeWithReason(CallHangup, "cancelled", reason)
 }
 
 func (c *inboundCall) closeWithHangup() {
+	var reason ReasonHeader
+	if p := c.closeReason.Load(); p != nil {
+		reason = *p
+	}
+	c.closeWithReason(CallHangup, "hangup", reason)
+}
+
+func (c *inboundCall) closeWithReason(status CallStatus, reasonName string, reason ReasonHeader) {
 	c.state.DeferUpdate(func(info *livekit.SIPCallInfo) {
 		info.DisconnectReason = livekit.DisconnectReason_CLIENT_INITIATED
+		if info.Error == "" {
+			if !reason.IsZero() {
+				info.Error = reason.String()
+			}
+		}
 	})
-	c.close(false, CallHangup, "hangup")
+	if reason.Type != "" {
+		change := true
+		switch reason.Type {
+		case "Q.850":
+			switch reason.Cause {
+			case 16: // Normal call clearing
+				change = false
+			}
+		}
+		if change {
+			reasonName = fmt.Sprintf("bye-%s-%d", strings.ToLower(reason.Type), reason.Cause)
+		}
+	}
+	c.close(false, status, reasonName)
```

Comment:
should the first argument be false in all cases ?

Reply:
Yes, we got a response from the remote, so it's not a critical error to log.